admin_manual/enterprise/external_storage/windows-network-drive_configuration.rst

```diff
@@ -144,247 +144,229 @@ ownCloud supports SMB notifications with an ``occ`` command, ``occ wnd:listen``.
+The listener stores the notifications.
+Any change in the mount point configuration, such as adding or removing new mounts, and logins by new users, won't affect the behavior, so there is no need to restart the listener in those cases.
 
-Running the WND Listener as a Service
--------------------------------------
+In case you have several mount point configurations, note that each listener attaches to one host and share.
+If there are several mount configurations targeting different shares, you'll need to spawn one listener for each.
+For example, if you have one configuration with ``10.0.0.2/share1`` and another with ``10.0.0.2/share2``, you'll need to spawn 2 listeners, one for the first configuration and another for the second.
 
-There are several different approaches available to running the Windows Network Drive listener as a service.
+Third, run the ``wnd:process-queue`` periodically, usually via :ref:`a Cron job <cron_job_label>`.
+The command processes all the stored notifications for a specific host and share.
+If you have several, you could set up several Cron jobs, one for each host and share with different intervals, depending on the load or update urgency.
+As a simple example, you could run the command every 2 minutes for one server and every 5 minutes for another.
 
-As a Cron Job
-~~~~~~~~~~~~~
+As said, the command processes all the stored notifications, squeeze them and scan the resulting folders.
+The process might crash if there are too many notifications, or if it has too many storages to update.
+The ``--chunk-size`` option will help by making the command process all the notifications in buckets of that size.
 
-Firstly, create a new script called ``wnd-listen.sh`` and add the code below to it, adjusting the path to your ownCloud installation so that it’s specific to your installation.
+On the one hand the memory usage is reduced, on the other hand there is more network activity.
+We recommend using the option with a value high enough to process a large number of notifications, but not so large to crash the process.
+Between 200 and 500 should be fine, and we'll likely process all the notifications in one go.
 
-.. code-block:: bash
+Optimizing wnd:process-queue
+----------------------------
 
-   #!/bin/bash 
-   until php -f /var/www/owncloud/occ wnd:listen $@; do
-      echo "occ wnd:listen crashed with exit code $?.  Respawning.." >&2
-      sleep 1
-   done
+``wnd:process-queue`` creates all the storages that need to be updated from scratch.
+To do so, we need to fetch all the users from all the backends (currently only the ones that have logged in at least once because the others won't have the storages that we'll need updates)
 
-Then, make the script executable and ensure that it is owned by your HTTP user. 
-To do that, run the following commands, changing ``<HTTP_USER>`` as required.
+To optimize this, ``wnd:process-queue`` make use of two switches: "--serializer-type" and "--serializer-params".
```

Comment:
I'd add a note to not use this option if the process-queue is fast enough without it. The option has some drawbacks specially regarding password changes in the backend (AD or whatever system Windows is using)